src/memspaces/memspace_host_all.c

```diff
@@ -40,7 +40,10 @@ static umf_result_t umfMemspaceHostAllCreate(umf_memspace_handle_t *hMemspace) {
     // object that can be present on multiple levels.
     // Source: https://www.open-mpi.org/projects/hwloc/doc/hwloc-v2.10.0-letter.pdf
     int nNodes = hwloc_get_nbobjs_by_type(topology, HWLOC_OBJ_NUMANODE);
-    assert(nNodes != -1);
+    if (nNodes < 0) {
+        umf_ret = UMF_RESULT_ERROR_UNKNOWN;
```

Comment:
is there anything we can print here, perhaps...?

Reply:
AFAIK, the logger is not ready yet.

Reply:
This should be done for each error in this file (and the whole project), in the separate PR. This PR is about fixing the Coverity issue.